R/indexing.R

```diff
@@ -53,6 +53,55 @@
 # - G[1:3,2,eid=TRUE]
 #               create an edge sequence
 
+get_adjacency_submatrix <- function(x, i, j, attr = NULL, sparse = TRUE) {
+  # If i or j is NULL, assume all nodes
+  # if not NULL make sure to handle duplicates correctly
+  if (missing(i)) {
+    i <- seq_len(vcount(x))
+  }
+  if (missing(j)) {
+    j <- seq_len(vcount(x))
+  }
+
+  adj <- adjacent_vertices(x, i, mode = "out")
+  i_degree <- map_int(adj, length)
+
+  from_id <- rep(i, i_degree)
+  to_id <- unlist(adj)
+
+  edge_list <- data.frame(from = as.integer(from_id), to = as.integer(to_id))
+  edge_list <- edge_list[edge_list$to %in% j, ]
+
+  row_indices <- edge_list[[1]]
+  col_indices <- edge_list[[2]]
+
+  values <- if (is.null(attr)) {
+    1
+  } else {
+    valid_edges <- get_edge_ids(x, edge_list)
+    edge_attr(x, attr, valid_edges)
+  }
+
+
+  res <- Matrix::sparseMatrix(
+    i = match(row_indices, i),
+    j = match(col_indices, j),
+    x = values,
+    dims = c(length(i), length(j))
+  )
+
+  if (!sparse) {
+    res <- as.matrix(res)
+  }
```

Comment:
Move this (and the arg) out of this function, and un-sparsify after the telescope?